tests/test_engage.py

```diff
@@ -99,6 +99,21 @@ def test_git_query(git_repo, capsys, workspace, monkeypatch):
     with pytest.raises(TypeError):
         git_query()
 
+def test_git_query_new_repo(git_repo_no_commits, capsys, workspace, monkeypatch):
+
+    # Check that git_query correctly determines that the repo is not currently clean
+    assert git_query(git_repo_no_commits, "catalogue_results") == False
```

Comment:
Minor stylistic point: `assert not git_query(git_repo_no_commits, "catalogue_results")`